### PR TITLE
#162948806 Get all articles redux

### PR DESCRIPTION
**Pull request by Georgeygigz**
**What does this PR do?**

* It creates functionality for listing all articles

**Description of Task to be completed?**

* Adding functionality for listing all articles

**How should this be manually tested?**

* clone this repo `https://github.com/andela/ah-django-unchained-frontend`
* `cd ah-django-unchained-frontend`
* `git checkout ft-list-articles-162948806`
* Install the required dependencies `npm install`
* `npm start to run the local server`

**Pivotal Tracker story**
[#162948804](https://www.pivotaltracker.com/story/show/162948804)

**Screenshot**
image

### CODE REVIEW

File: src/pages/AllArticles/allarticles.test.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import { shallow, mount } from 'enzyme';
+import jest from 'jest-mock';
+import { AllArticles, mapStateToProps } from './index';
+
+describe('Get All articles', () => {
+  let props;
+  let wrapper;
+  let state;
+  let wrapperInstance;
+  beforeEach(() => {
```
Comment (kwameasiago) [inline src/pages/AllArticles/allarticles.test.js]: Add a line for readability

File: src/store/modules/getArticles/types.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export const GET_ALL_ARTICLES_REQUEST = 'GET_ALL_ARTICLES_REQUEST';
+export const GET_ALL_ARTICLES_SUCCESS = 'GET_ALL_ARTICLES_SUCCESS';
+export const GET_ALL_ARTICLES_FAILURE = 'GET_ALL_ARTICLES_FAILURE';
```
Comment (kwameasiago) [inline src/store/modules/getArticles/types.js]: Add a line at the end of the file

**Comment by andrewhingah** [comment]
Good work George. However, you can change the body of your commit message from `Write test for getting functionality` to `Write test for getting all articles`. This will be more descriptive.

File: src/pages/AllArticles/allarticles.test.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { shallow, mount } from 'enzyme';
+import jest from 'jest-mock';
+import { AllArticles, mapStateToProps } from './index';
+
+describe('Get All articles', () => {
```
Comment (andrewhingah) [inline src/pages/AllArticles/allarticles.test.js]: I think the title of the describe block should be the name of the component. For instance, `AllArticles` and not `Get All Articles`. But @cosmas28 can confirm this.

Comment (cosmas28) [reply src/pages/AllArticles/allarticles.test.js]: I agree with @andrewhingah

Comment (Georgeygigz) [reply src/pages/AllArticles/allarticles.test.js]: I have already implemented this.

File: src/components/ArticlePanel/index.jsx [hunk 4]
```diff
@@ -1,30 +1,11 @@
...
-            </a>
-            <a href="/" className="card-link btn btn-info">
-              <FaThumbsDown />
-            </a>
-            <a href="/" className="card-link  btn btn-info">Read</a>
-            <a href="/" className="card-link  btn btn-info">Bookmark</a>
-          </div>
-        </div>
-        <br />
+        <AllArticles />
```
Comment (cosmas28) [inline src/components/ArticlePanel/index.jsx]: - It's not a good practice to nest container component to a presentational component.
- Kindly undo these changes and pass props to replace the dummy data in the `ArticlePanel` component.

File: src/pages/AllArticles/index.jsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+import { FaStar, FaThumbsUp, FaThumbsDown } from 'react-icons/fa';
+import { connect } from 'react-redux';
+import { getallArticles } from '../../store/modules/getArticles/index';
+import Loader from '../../components/Loader/index';
+
+export class AllArticles extends Component {
```
Comment (cosmas28) [inline src/pages/AllArticles/index.jsx]: I don't think the `AllArticles` component is needed. Use the home `HomaPage` component to display all the articles. Kindly take time to learn the code base carefully reuse the available properly. Sync with @SelaDanti to explain the use of the `HomePage` and `ArticlePanel` components.

**Review by cosmas28 — CHANGES_REQUESTED** [review]

File: src/components/ArticlePanel/index.jsx (hunk 4 above)
Comment (Georgeygigz) [reply src/components/ArticlePanel/index.jsx]: Okay will work on this

File: src/pages/AllArticles/index.jsx (hunk 5 above)
Comment (Georgeygigz) [reply src/pages/AllArticles/index.jsx]: Okay thanks, working on this.

File: src/store/modules/getArticles/getarticles.actions.test.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
+import moxios from 'moxios';
+import { GET_ALL_ARTICLES_SUCCESS, GET_ALL_ARTICLES_REQUEST } from './types';
```
Comment (cosmas28) [inline src/store/modules/getArticles/getarticles.actions.test.js:2]: I like the way you named this action. Keep it up.

File: src/store/modules/getArticles/index.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
+import * as types from './types';
```
Comment (cosmas28) [inline src/store/modules/getArticles/index.js:1]: In the file, I also like the way you named your actions. Keep it up.

File: src/store/modules/getArticles/getarticles.actions.test.js (hunk 6 above)
Comment (Georgeygigz) [reply src/store/modules/getArticles/getarticles.actions.test.js:2]: Thanks, I will work hard to stick to this

File: src/store/modules/getArticles/index.js (hunk 7 above)
Comment (Georgeygigz) [reply src/store/modules/getArticles/index.js:1]: Thanks, I will try and stick to this too

File: src/pages/HomePage/index.jsx [hunk 8]
```diff
@@ -1,14 +1,47 @@
...
+export class HomePage extends Component {
+  componentDidMount() {
+    const { getallArticles } = this.props;
+    getallArticles();
+  }
+
+  render() {
+    const { message } = this.props;
+    const articles = this.props;
+    const articlesList = articles.articles.length ? (
```
Comment (cosmas28) [inline src/pages/HomePage/index.jsx]: Two feedback about this logic here:
- It is not a good practice to have this condition here, so please do this in the `return` statement.
- the `No articles` message is displayed even when the page is loading, I think it should only show up when the loader has stopped.

**Review by cosmas28 — CHANGES_REQUESTED** [review]

File: src/pages/HomePage/index.jsx (hunk 8 above)
Comment (Georgeygigz) [reply src/pages/HomePage/index.jsx]: Okay I will work on this

**Review by cosmas28 — APPROVED** [review]

**Pull request merged**